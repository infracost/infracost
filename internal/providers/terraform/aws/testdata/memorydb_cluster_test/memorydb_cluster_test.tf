provider "aws" {
  region                      = "us-east-1"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  skip_region_validation      = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

resource "aws_memorydb_cluster" "redis" {
  name               = "my-cluster"
  node_type          = "db.r6g.large"
  num_shards         = 2
  num_replicas_per_shard = 1
  engine             = "redis"
  snapshot_retention_limit = 2
}

resource "aws_memorydb_cluster" "valkey" {
  name               = "my-valkey-cluster"
  node_type          = "db.r6g.large"
  num_shards         = 2
  num_replicas_per_shard = 1
  engine             = "valkey"
  snapshot_retention_limit = 2
}

resource "aws_memorydb_cluster" "redis_with_usage" {
  name               = "my-cluster-with-usage"
  node_type          = "db.r6g.large"
  num_shards         = 2
  num_replicas_per_shard = 1
  engine             = "redis"
  snapshot_retention_limit = 2
}

resource "aws_memorydb_cluster" "valkey_with_usage" {
  name               = "my-valkey-cluster-with-usage"
  node_type          = "db.r6g.large"
  num_shards         = 2
  num_replicas_per_shard = 1
  engine             = "valkey"
  snapshot_retention_limit = 2
}

resource "aws_memorydb_cluster" "redis_reserved" {
  name               = "my-reserved-cluster"
  node_type          = "db.r6g.large"
  num_shards         = 2
  num_replicas_per_shard = 1
  engine             = "redis"
  snapshot_retention_limit = 2
}

resource "aws_appautoscaling_target" "memorydb_shards" {
  max_capacity       = 4
  min_capacity       = 2
  resource_id        = "memorydb:my-autoscaling-cluster"
  scalable_dimension = "memorydb:cluster:Shards"
  service_namespace  = "memorydb"
}

resource "aws_appautoscaling_target" "memorydb_replicas" {
  max_capacity       = 2
  min_capacity       = 1
  resource_id        = "memorydb:my-autoscaling-cluster"
  scalable_dimension = "memorydb:cluster:ReplicasPerShard"
  service_namespace  = "memorydb"
}

resource "aws_memorydb_cluster" "autoscaling" {
  name               = "my-autoscaling-cluster"
  node_type          = "db.r6g.large"
  num_shards         = 2
  num_replicas_per_shard = 1
  engine             = "redis"
  snapshot_retention_limit = 2
}
