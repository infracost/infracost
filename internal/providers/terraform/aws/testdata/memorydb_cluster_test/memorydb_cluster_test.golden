{
  "version": "0.2",
  "metadata": {
    "infracostCommand": "breakdown",
    "vcsBranch": "stub",
    "vcsCommitSha": "stub",
    "vcsCommitAuthorName": "stub",
    "vcsCommitAuthorEmail": "stub",
    "vcsCommitTimestamp": "stub",
    "vcsCommitMessage": "stub",
    "vcsRepositoryUrl": "stub"
  },
  "currency": "USD",
  "projects": [
    {
      "name": "memorydb_cluster_test",
      "metadata": {
        "path": "memorydb_cluster_test",
        "type": "terraform_dir",
        "vcsSubPath": "stub"
      },
      "pastBreakdown": {
        "resources": [],
        "totalHourlyCost": "0",
        "totalMonthlyCost": "0"
      },
      "breakdown": {
        "resources": [
          {
            "name": "aws_memorydb_cluster.autoscaling",
            "metadata": {},
            "hourlyCost": "2.4",
            "monthlyCost": "1752",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, on-demand, db.r6g.large, autoscaling)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.4",
                "hourlyCost": "2.4",
                "monthlyCost": "1752"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.2",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.redis",
            "metadata": {},
            "hourlyCost": "2.4",
            "monthlyCost": "1752",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.4",
                "hourlyCost": "2.4",
                "monthlyCost": "1752"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.2",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.redis_reserved",
            "metadata": {},
            "hourlyCost": "0",
            "monthlyCost": "1022",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, reserved, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.233333",
                "hourlyCost": "1.4",
                "monthlyCost": "1022"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.2",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.redis_with_usage",
            "metadata": {},
            "hourlyCost": "2.4",
            "monthlyCost": "1954.1",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.4",
                "hourlyCost": "2.4",
                "monthlyCost": "1752"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "1.3698630137",
                "monthlyQuantity": "1000",
                "price": "0.2",
                "hourlyCost": "0.2739726027",
                "monthlyCost": "200"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0.1369863014",
                "monthlyQuantity": "100",
                "price": "0.021",
                "hourlyCost": "0.0028767123",
                "monthlyCost": "2.1"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.valkey",
            "metadata": {},
            "hourlyCost": "1.68",
            "monthlyCost": "1226.4",
            "costComponents": [
              {
                "name": "MemoryDB instance (valkey, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.28",
                "hourlyCost": "1.68",
                "monthlyCost": "1226.4"
              },
              {
                "name": "Data written (free up to 10TB)",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.valkey_with_usage",
            "metadata": {},
            "hourlyCost": "1.68",
            "monthlyCost": "1426.5",
            "costComponents": [
              {
                "name": "MemoryDB instance (valkey, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.28",
                "hourlyCost": "1.68",
                "monthlyCost": "1226.4"
              },
              {
                "name": "Data written (over 10TB)",
                "unit": "GB",
                "hourlyQuantity": "6.5068493151",
                "monthlyQuantity": "4760",
                "price": "0.04",
                "hourlyCost": "0.2602739726",
                "monthlyCost": "190.4"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0.2739726027",
                "monthlyQuantity": "200",
                "price": "0.021",
                "hourlyCost": "0.0057534247",
                "monthlyCost": "4.2"
              }
            ]
          }
        ],
        "totalHourlyCost": "10.56",
        "totalMonthlyCost": "7711"
      },
      "diff": {
        "resources": [
          {
            "name": "aws_memorydb_cluster.autoscaling",
            "metadata": {},
            "hourlyCost": "2.4",
            "monthlyCost": "1752",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, on-demand, db.r6g.large, autoscaling)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.4",
                "hourlyCost": "2.4",
                "monthlyCost": "1752"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.2",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.redis",
            "metadata": {},
            "hourlyCost": "2.4",
            "monthlyCost": "1752",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.4",
                "hourlyCost": "2.4",
                "monthlyCost": "1752"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.2",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.redis_reserved",
            "metadata": {},
            "hourlyCost": "0",
            "monthlyCost": "1022",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, reserved, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.233333",
                "hourlyCost": "1.4",
                "monthlyCost": "1022"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.2",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.redis_with_usage",
            "metadata": {},
            "hourlyCost": "2.4",
            "monthlyCost": "1954.1",
            "costComponents": [
              {
                "name": "MemoryDB instance (redis, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.4",
                "hourlyCost": "2.4",
                "monthlyCost": "1752"
              },
              {
                "name": "Data written",
                "unit": "GB",
                "hourlyQuantity": "1.3698630137",
                "monthlyQuantity": "1000",
                "price": "0.2",
                "hourlyCost": "0.2739726027",
                "monthlyCost": "200"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0.1369863014",
                "monthlyQuantity": "100",
                "price": "0.021",
                "hourlyCost": "0.0028767123",
                "monthlyCost": "2.1"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.valkey",
            "metadata": {},
            "hourlyCost": "1.68",
            "monthlyCost": "1226.4",
            "costComponents": [
              {
                "name": "MemoryDB instance (valkey, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.28",
                "hourlyCost": "1.68",
                "monthlyCost": "1226.4"
              },
              {
                "name": "Data written (free up to 10TB)",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0",
                "hourlyCost": "0",
                "monthlyCost": "0"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0",
                "monthlyQuantity": "0",
                "price": "0.021",
                "hourlyCost": "0",
                "monthlyCost": "0"
              }
            ]
          },
          {
            "name": "aws_memorydb_cluster.valkey_with_usage",
            "metadata": {},
            "hourlyCost": "1.68",
            "monthlyCost": "1426.5",
            "costComponents": [
              {
                "name": "MemoryDB instance (valkey, on-demand, db.r6g.large)",
                "unit": "hours",
                "hourlyQuantity": "6",
                "monthlyQuantity": "4380",
                "price": "0.28",
                "hourlyCost": "1.68",
                "monthlyCost": "1226.4"
              },
              {
                "name": "Data written (over 10TB)",
                "unit": "GB",
                "hourlyQuantity": "6.5068493151",
                "monthlyQuantity": "4760",
                "price": "0.04",
                "hourlyCost": "0.2602739726",
                "monthlyCost": "190.4"
              },
              {
                "name": "Backup storage",
                "unit": "GB",
                "hourlyQuantity": "0.2739726027",
                "monthlyQuantity": "200",
                "price": "0.021",
                "hourlyCost": "0.0057534247",
                "monthlyCost": "4.2"
              }
            ]
          }
        ],
        "totalHourlyCost": "10.56",
        "totalMonthlyCost": "7711"
      },
      "summary": {
        "totalDetectedResources": 8,
        "totalSupportedResources": 6,
        "totalUnsupportedResources": 0,
        "totalUsageBasedResources": 6,
        "totalNoPriceResources": 0,
        "unsupportedResourceCounts": {},
        "noPriceResourceCounts": {}
      }
    }
  ],
  "totalHourlyCost": "10.56",
  "totalMonthlyCost": "7711",
  "pastTotalHourlyCost": "0",
  "pastTotalMonthlyCost": "0",
  "diffTotalHourlyCost": "10.56",
  "diffTotalMonthlyCost": "7711",
  "timeGenerated": "REPLACED_TIME",
  "summary": {
    "totalDetectedResources": 8,
    "totalSupportedResources": 6,
    "totalUnsupportedResources": 0,
    "totalUsageBasedResources": 6,
    "totalNoPriceResources": 0,
    "unsupportedResourceCounts": {},
    "noPriceResourceCounts": {}
  }
}
